### PR TITLE
Resend envelope if we sent it but didn't get an ack

### PR DESCRIPTION
**Pull request by dshulyak**
TBD

### CODE REVIEW

File: services/shhext/envelopes.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+		whisper.EventBatchAcknowledged: mon.handleAcknowledgedBatch,
+	}
+
+	if handler, ok := handlers[event.Event]; ok {
+		handler(event)
+	}
+}
+
+func (mon *EnvelopesMonitor) handleEventEnvelopeSent(event whisper.EnvelopeEvent) {
+	if mon.mailServerConfirmation {
```
Comment (mandrigin) [inline services/shhext/envelopes.go:102]: what if this is `false`?

File: services/shhext/envelopes.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+func (mon *EnvelopesMonitor) checkResendEnvelope(hash common.Hash) {
+	mon.mu.Lock()
+	defer mon.mu.Unlock()
+	state, ok := mon.cache[hash]
+	if !ok || state.state == EnvelopeSent {
+		return
+	}
+	state.expired++
+	if state.expired == mon.config.MaxRetries {
```
Comment (mandrigin) [inline services/shhext/envelopes.go:145]: `>=` here

File: services/shhext/envelopes.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+		log.Debug("maximum number of retries reached", "hash", hash)
```
Comment (mandrigin) [inline services/shhext/envelopes.go:146]: shouldn't we just `return` here?

**Review by mandrigin — CHANGES_REQUESTED** [review]
in general, looks good. I have a few questions on logic in a few places.

File: services/shhext/envelopes.go (hunk 1 above)
Comment (dshulyak) [reply services/shhext/envelopes.go:102]: this code was moved due to refactoring. but in general this only ensures that we accept this event only when envelope was sent to mail server

**Comment by dshulyak** [comment]
this needs to be implemented differently. we need to wrap original payload and create envelope every time we do a retry. when envelope is sealed the deadline is sealed as well, so we need to adjust it every time. i will create a separate issue

**Pull request closed without merging**